plugins/com.google.cloud.tools.eclipse.appengine.deploy/src/com/google/cloud/tools/eclipse/appengine/deploy/WarPublisher.java

```diff
@@ -71,12 +70,7 @@ private static void publish(IProject project, IPath destination, boolean explode
     progress.setTaskName(Messages.getString("task.name.publish.war"));
 
     PublishHelper publishHelper = new PublishHelper(null);
-    J2EEFlexProjDeployable deployable =
-        new J2EEFlexProjDeployable(project, ComponentCore.createComponent(project));
-
-    List<IModuleResource> resourcesToDeploy = Lists.newArrayList(deployable.members());
-    resourcesToDeploy.addAll(computeWebFragmentJarResources(deployable, progress));
-    IModuleResource[] resources = resourcesToDeploy.toArray(new IModuleResource[0]);
+    IModuleResource[] resources = flattenResources(project, progress);
 
     if (exploded) {
       publishHelper.publishFull(resources, destination, progress.newChild(100));
```

Comment:
Was there a difference between `publishFull` and `publishSmart`?  Or did you just figure that since we're deploying to a new directory each time, there's little point with the `publishSmart`?